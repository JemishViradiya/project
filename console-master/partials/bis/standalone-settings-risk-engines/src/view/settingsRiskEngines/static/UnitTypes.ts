const UnitTypes = Object.freeze({
  miles: 'miles',
  yards: 'yards',
  kilometers: 'kilometers',
  meters: 'meters',
})

export default UnitTypes
