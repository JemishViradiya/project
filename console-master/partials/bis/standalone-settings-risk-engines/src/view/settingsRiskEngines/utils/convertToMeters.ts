import UnitTypes from '../static/UnitTypes'

const YARD_IN_METER = 0.9144
const METERS_IN_KILOMETER = 1000
const YARDS_IN_MILE = 1760

const isNil = value => value == null

const validate = (value, unit) => {
  if (isNil(value) || isNil(unit) || !Object.values(UnitTypes).includes(unit)) {
    throw new Error(`Unable to make conversion to meters. Input: value=${value}, unit=${unit}.`)
  }
}

export default (value, unit) => {
  validate(value, unit)

  switch (unit) {
    case UnitTypes.miles:
      return value * YARDS_IN_MILE * YARD_IN_METER
    case UnitTypes.kilometers:
      return value * METERS_IN_KILOMETER
    case UnitTypes.yards:
      return value * YARD_IN_METER
    default:
      return value
  }
}
