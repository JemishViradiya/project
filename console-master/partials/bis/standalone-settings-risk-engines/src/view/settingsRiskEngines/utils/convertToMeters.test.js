import UnitTypes from '../static/UnitTypes'
import convertToMeters from './convertToMeters'

describe('convertToMeters', () => {
  it.each`
    value      | unit                    | expectedResult
    ${253542}  | ${UnitTypes.miles}      | ${408036296.448}
    ${9236}    | ${UnitTypes.yards}      | ${8445.3984}
    ${43636}   | ${UnitTypes.kilometers} | ${43636000}
    ${9236478} | ${UnitTypes.meters}     | ${9236478}
  `('should return $expectedResult when when input value is $value $unit', ({ value, unit, expectedResult }) => {
    // when
    const result = convertToMeters(value, unit)

    // then
    expect(result).toBe(expectedResult)
  })

  it.each`
    value        | unit
    ${undefined} | ${undefined}
    ${null}      | ${null}
    ${6345}      | ${undefined}
    ${undefined} | ${UnitTypes.miles}
    ${31232}     | ${'SOME_OTHER_UNIT'}
  `('should throw error matching snapshot when value is $value and unit is $unit', ({ value, unit }) => {
    expect(() => {
      convertToMeters(value, unit)
    }).toThrowErrorMatchingSnapshot()
  })
})
