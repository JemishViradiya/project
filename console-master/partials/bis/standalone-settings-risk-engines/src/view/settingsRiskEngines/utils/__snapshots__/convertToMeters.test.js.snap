// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convertToMeters should throw error matching snapshot when value is 6345 and unit is undefined 1`] = `"Unable to make conversion to meters. Input: value=6345, unit=undefined."`;

exports[`convertToMeters should throw error matching snapshot when value is 31232 and unit is SOME_OTHER_UNIT 1`] = `"Unable to make conversion to meters. Input: value=31232, unit=SOME_OTHER_UNIT."`;

exports[`convertToMeters should throw error matching snapshot when value is null and unit is null 1`] = `"Unable to make conversion to meters. Input: value=null, unit=null."`;

exports[`convertToMeters should throw error matching snapshot when value is undefined and unit is miles 1`] = `"Unable to make conversion to meters. Input: value=undefined, unit=miles."`;

exports[`convertToMeters should throw error matching snapshot when value is undefined and unit is undefined 1`] = `"Unable to make conversion to meters. Input: value=undefined, unit=undefined."`;
